osquery/tables/system/darwin/user_groups.mm

```diff
@@ -48,38 +49,43 @@ void genODEntries(ODRecordType type, std::set<std::string>& names) {
     return;
   }
 
+  NSError* attrErr = nullptr;
+  // if IsHidden does not exist or has an invalid value it's equivalent
+  // to IsHidden: 0
+  std::string isHidden;
+
   for (ODRecord* re in od_results) {
-    names.insert([[re recordName] UTF8String]);
+    auto isHiddenValue = [re valuesForAttribute:@"dsAttrTypeNative:IsHidden"
+                                          error:&attrErr];
+
+    // set isHidden back to 0 before processing atrribute
+    isHidden = "0";
+    if (isHiddenValue.count >= 1) {
+      isHidden = std::string([isHiddenValue[0] UTF8String]);
+      boost::algorithm::to_lower(isHidden);
+      // isHiddenValue can be 1 || True || YES to hide the user or group
+      if (isHidden == "yes" || isHidden == "1" || isHidden == "true") {
```

Comment:
Would it be better to use `tryTo<bool>()` from `osquery/utils/conversions/tryto.h` instead of manual parsing?
Something like:
```c++
tryTo<bool>(std::string([isHiddenValue[0] UTF8String])).takeOr(false);
```